### PR TITLE
Rust: Vendor build files for 3rdparty dependencies.

### PR DESCRIPTION
**Pull request by criemen**
We've been observing some performance issues using `crate_universe` on CI. Therefore, we're moving to vendor the auto-generated BUILD files in our repository. This should provide a nice speed boost, while getting rid of the complexity of the "rust cache" job we've been using when we had a lot of git dependencies.

This PR includes a vendor script, and I'll put up a CI job internally that runs that vendor script on `Cargo.toml` and `Cargo.lock` changes, to check that the vendored files are in sync.

For rust on linux this saves >1min on CI, macos also ca. 1min, Windows 3min.
I assume similar time savings hold for Ruby, and a bit less for python.

### CODE REVIEW

File: python/extractor/tsg-python/tsp/Cargo.toml [hunk 1]
```diff
@@ -24,6 +24,7 @@ include = [
 [lib]
 path = "bindings/rust/lib.rs"
 
+## When updating these dependencies, run `misc/bazel/3rdparty/update_cargo_deps.sh`
```
Comment (tausbn) [inline python/extractor/tsg-python/tsp/Cargo.toml:27]: This one gets a double comment! I assume there's no semantic significance. 🙂

**Review by tausbn — APPROVED** [review]
Python bits LGTM. 👍

File: python/extractor/tsg-python/tsp/Cargo.toml (hunk 1 above)
Comment (criemen) [reply python/extractor/tsg-python/tsp/Cargo.toml:27]: Does it? I tried to add the comment to every `Cargo.toml` file that contributes to an extractor build, with the exception of ql-for-ql, which we don't build with bazel.

Comment (tausbn) [reply python/extractor/tsg-python/tsp/Cargo.toml:27]: I meant compared to the other `Cargo.toml` file, where the added line starts with a single comment character. Sorry for the confusing wording.

Comment (criemen) [reply python/extractor/tsg-python/tsp/Cargo.toml:27]: Ah, double shebang! Yeah that wasn't intentional, but shouldn't change anything either.

**Review by oscarsj — APPROVED** [review]
LGTM, at least internally consisten - some of the Bazel workings might be lost to me but paths match, moved code makes sense, and bash scripts seem correct, so :shipit:

**Pull request merged**